gitter/rest.go

```diff
@@ -63,16 +62,15 @@ func (client *RestAPIClient) Get(ctx context.Context, resourceFragments []string
 	// Do request
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed executing HTTP request: %w", err)
 	}
 
 	defer resp.Body.Close()
 
 	// Handle response
 	err = json.NewDecoder(resp.Body).Decode(&intf)
 	if err != nil {
-		log.Errorf("Can not unmarshal given JSON structure: %+v", err)
```

Comment:
Instead of logging in each client module, return the error with extra information.
The calling method can judge the severity of the error state and then determine the appropriate logging level, if required.